src/calculateRentalCost.js

```diff
@@ -3,8 +3,24 @@
  *
  * @return {number}
  */
-function calculateRentalCost(days) {
-  // write code here
+const DAILY_RATE = 40;
+const LONG_TERM_DAYS = 7;
+const LONG_TERM_DISCOUNT = 50;
+const MEDIUM_TERM_DAYS = 3;
+const MEDIUM_TERM_DISCOUNT = 20;
+
+function calculateRentalCost(numberOfDays) {
+  const basePrice = numberOfDays * DAILY_RATE;
+
+  if (numberOfDays >= LONG_TERM_DAYS) {
+    return basePrice - LONG_TERM_DISCOUNT;
+  }
+
+  if (numberOfDays >= MEDIUM_TERM_DAYS) {
+    return basePrice - MEDIUM_TERM_DISCOUNT;
```

Comment:
Logical error: The medium-term discount is applied even if the long-term discount condition is met. To fix this, the long-term discount condition should be checked first, as it currently is, but the medium-term discount should only be checked if the long-term condition is not met. Consider using an `else if` statement here to ensure that only one discount is applied.